### PR TITLE
feat: Add mask-aware dataset grids for instance segmentation label validation

### PR DESCRIPTION
**Pull request by pirnerjonas**
## Summary
- render instance masks in saved dataset grids when targets include `masks`
- pass masks through `supervision.Detections` and draw them before boxes and labels
- add a regression test that verifies segmentation masks are forwarded to the mask annotator
- use Matplotlib's non-interactive `Agg` backend for dataset grid file rendering so CI and headless environments do not require Tk

Refs #1013

## Validation
- `uv run --no-sync pytest tests/datasets/test_save_grids.py -q`
- `MPLBACKEND=TkAgg uv run --no-sync pytest tests/datasets/test_save_grids.py::test_save_grid_passes_masks_to_supervision -q`
- `uvx pre-commit run --all-files`
- GitHub PR checks passed on `f5e5679`, including `Testing (windows-latest, 3.10)`

### CODE REVIEW

**Comment by ZubairAli999** [comment]
@pirnerjonas  If you need any help, just let me know. I'm available!

**Comment by Borda** [comment]
sample code:
```
import json
from pathlib import Path

import numpy as np
import torch
import torchvision.transforms as T  # noqa: N812
from PIL import Image

from rfdetr.datasets.coco import convert_coco_poly_to_mask
from rfdetr.datasets.save_grids import DatasetGridSaver
from rfdetr.visualize.training import _ensure_headless_backend

_DATA_DIR = Path(__file__).parent.parent / "data"
_IMAGE_FILE = "000000579321.jpg"  # dog + person, 2 clean segmented instances
# Must match DatasetGridSaver's inv_normalize so colors round-trip correctly for display.
_NORMALIZE = T.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])


def _load_coco_sample() -> tuple[torch.Tensor, torch.Tensor, torch.Tensor, torch.Tensor]:
    """Load one real COCO val2017 image plus its ground-truth instance masks and boxes."""
    with (_DATA_DIR / "annotations" / "instances_val2017.json").open() as f:
        coco = json.load(f)
    img_info = next(im for im in coco["images"] if im["file_name"] == _IMAGE_FILE)
    anns = [
        a
        for a in coco["annotations"]
        if a["image_id"] == img_info["id"] and a.get("segmentation") and a.get("iscrowd", 0) == 0
    ]

    pil_image = Image.open(_DATA_DIR / "val2017" / _IMAGE_FILE).convert("RGB")
    width, height = pil_image.size

    masks = convert_coco_poly_to_mask([a["segmentation"] for a in anns], height, width).bool()
    boxes = torch.tensor(
        [
            [
                (a["bbox"][0] + a["bbox"][2] / 2) / width,
                (a["bbox"][1] + a["bbox"][3] / 2) / height,
                a["bbox"][2] / width,
                a["bbox"][3] / height,
            ]
            for a in anns
        ]
    )
    labels = torch.tensor([a["category_id"] for a in anns])

    image_tensor = torch.from_numpy(np.array(pil_image)).permute(2, 0, 1).float() / 255.0
    return _NORMALIZE(image_tensor), masks, boxes, labels


def main() -> None:
    import matplotlib

    _ensure_headless_backend(matplotlib)
    import matplotlib.pyplot as plt
    import supervision as sv

    image, masks, boxes, labels = _load_coco_sample()
    size = torch.tensor([image.shape[1], image.shape[2]])

    inv_normalize = T.Normalize(
        mean=[-0.485 / 0.229, -0.456 / 0.224, -0.406 / 0.225],
        std=[1 / 0.229, 1 / 0.224, 1 / 0.225],
    )
    mask_annotator = sv.MaskAnnotator(opacity=0.45)
    box_annotator = sv.BoxAnnotator(thickness=2)
    label_annotator = sv.LabelAnnotator(text_color=sv.Color.BLACK, text_scale=0.5, text_padding=3)

    fig, axes = plt.subplots(1, 2, figsize=(10, 5))
    for ax, title, target_masks in [
        (axes[0], "Before PR #1014 — boxes only", None),
        (axes[1], "After PR #1014 — + instance masks", masks),
    ]:
        target = {"size": size, "boxes": boxes, "labels": labels}
        if target_masks is not None:
            target["masks"] = target_masks
        DatasetGridSaver._annotate_and_plot(
            image, target, ax, inv_normalize, mask_annotator, box_annotator, label_annotator
        )
        ax.set_title(title, fontsize=10)

    fig.suptitle(f"COCO val2017 — {_IMAGE_FILE}")
    fig.tight_layout()
    output_path = Path(__file__).parent / "output" / "mask_dataset_grid_demo" / "before_after_comparison.png"
    output_path.parent.mkdir(parents=True, exist_ok=True)
    fig.savefig(output_path, dpi=150)
    plt.close(fig)
    print(f"wrote: {output_path}")


if __name__ == "__main__":
    main()
```

image
